### DISCUSSION TITLE
Using parachain SDKs

### DISCUSSION THREAD

**Discussion by tad3j** [polkadot-api/polkadot-api#646] [General]
I've been evaluating this SDK (as a replacement of Polkadot SDK) to use on Apillon project and I like the benefits this SDK brings but I've noticed that it isn't compatible with most of the parachain SDKs, mostly due to the fact that all SDKs rely on Polkadot SDK (either they use it under the hood or require ApiPromise to be injected).

Here is an example of using Phala SDK and concrete issue where variable `testApi` has invalid type (TypedApi) passed to Phalas `OnChainRegistry`:
```
import { createClient } from 'polkadot-api';
import { getSmProvider } from 'polkadot-api/sm-provider';
import {  OnChainRegistry} from '@phala/sdk';

const smoldot = startFromWorker(
  new Worker(require.resolve('polkadot-api/smoldot/node-worker'), {
    stderr: true,
    stdout: true,
  }),
);
const client = createClient(getSmProvider(smoldot.addChain({ chainSpec })));
const testApi = client.getTypedApi(xyz);

const registry = await OnChainRegistry.create(testApi, {
  strategy: periodicityChecker(),
});
```

Most if not all SDKs all suffer from tight Polkadot SDK coupling with them that's why I wanted to open an issue to see what are other people thoughts on this issue. I know solution is not trivial since each of SDKs would have to support this SDK or at least use compatible interface. I'm afraid this issue will prevent adoption of this SDK since it's only compatible with generic substrate side but not with specialized parachain side.

Is it reasonable to expect that parachains will ever be compatible with different API clients?

**Comment by voliva** [comment]
Different libraries can choose what they want to build on top of other libraries.

Preexisting SDKs like phala's built on top of PolkadotJS will only work on top of PolkadotJS until they decide to build a different one on top of another client library. I don't think it would be feasible to have an abstraction on top of client libraries, since they all work differently.

Papi is now on 1.0.0, and we'll keep maintaining and adding in more features, but it's always going to be chain-agnostic. Having said that, as part of our 6-month roadmap, we will create SDKs for specific interactions within specific chains or pallets, published separately from polkadot-api. We'll focus on the ones that are more used globally, but they could also serve as examples for other parachain teams to build their own SDKs on top of that.

I'll move this issue into discussions, since there's no action to be done.

**Comment by tad3j** [comment]
My first thought was to have a unified interface across SDKs (currently class ApiPromise is hardcoded) since this would let developers use whichever client they want but on the other side I guess this would not be easy to achieve.

I feel like adoption would be much easier if parachain SDKs would be compatible with this new client but as far as I've talked with owners they have no plans to accommodate this.
